### DISCUSSION TITLE
'code': 'Unauthorized'

### DISCUSSION THREAD

**Discussion by ryankirkland** [saleweaver/python-amazon-sp-api#69] [FAQ]
I searched through the other threads covering this and nothing seems to be helping. I've followed the Selling Partner API official documentation for setting up the IAM user, policy, and role, as well as setting up STS. 

I've confirmed I am using the correct ID and Secret for the LWA fields and am using the AWS access and secret keys that are tied to the IAM user I created through the SP API docs. I am using the IAM Role ARN rather than the IAM user ARN per documentation recommendation.

I have all boxes checked under Roles in the App registration. I'm feeling at a loss for what else could possibly be wrong here. I am in the US region. My specific error message is:

[{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]

It says unauthorized, but I've generated my refresh token after selecting Authorize in Developer Central. Any guidance would be awesome - I am just trying to run the base call to the Orders API.

`# orders API
try:
    res = Orders().get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())
    print(res.payload)  # json data
except SellingApiException as ex:
    print(ex)`

**I was able to resolve this by just starting over with fresh IAM setup and registering a new client. I will update with the final reason why it was not working as soon as I figure it out myself in the hopes of preventing others from hitting the same thing!**

I will say - remember to register your app with the IAM Role ARN, not the IAM User ARN (though I was doing that in the first place, just felt that wasn't abundantly clear). Also, getting your refresh token is a function of Authorizing the app in Developer Central - something else I felt wasn't immediately clear and will hopefully save someone a couple minutes.

Thanks for putting this wrapper together though - loving it so far!

**Comment by diegosigu90** [comment]
Hi, I have the same issue, but when I try on postman I receive the data correctly. Any help?

thanks

**Reply by ryankirkland** [reply]
Are you getting the same error message I was? Can you provide how you're attempting your requests outside of Postman?

**Reply by diegosigu90** [reply]
This is my code. Im new in python so maybe im doing something wrong.


from sp_api.api import Orders
from sp_api.base import SellingApiException, Marketplaces
from datetime import date, datetime, timedelta
import json
import ast



marketplace_id = 'A1RKKUPIHCS9HS'


refresh_token = 'Atzr|IwEBIXXXXXXXXf'
lwa_client_secret = 'bd2XXXXXXX'
aws_access_key = 'AKIAXXXXXXXXX'
aws_secret_key = 'EbXXXXXXXXXXXXXXXX'
role_arn = 'arnXXXXXXXXXXXXXXX'

credentials = dict(
refresh_token=refresh_token, # Seller Central Dev Central Action/Authorize after account creation
lwa_app_id=lwa_client_id, # Seller Central
lwa_client_secret=lwa_client_secret,
aws_secret_key=aws_secret_key,
aws_access_key=aws_access_key,
role_arn=role_arn,
)

resp = Orders(credentials=credentials, marketplace=Marketplaces.ES).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())
print(resp)

**Comment by Priyankk18k** [comment]
Hi @ryankirkland,
I'm also getting the same message {'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}
I have also done everything from scratch but still getting the same message. Can you tell me what is the solution for this?
